Error building sample_code.cpp on Linux: undefined reference to `hmac_keccak_hash' [1 XMR]

Building sample_code.cpp on Linux results in a build error:

```
undefined reference to `hmac_keccak_hash'
```

This issue requests fixing the build error on Linux. It builds successfully on macOS.

To build the sample code, modify CMakeLists.txt with `set(BUILD_SAMPLE ON)` and follow the instructions to [build monero-cpp](https://github.com/woodser/monero-cpp?tab=readme-ov-file#using-monero-cpp-in-your-project).

That will generate a binary: `./build/sample_code`

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -412,7 +412,6 @@ if (BUILD_SAMPLE)
       ringct
       ringct_basic
       common
-      cncrypto
       blockchain_db
       blocks
       checkpoints
@@ -420,6 +419,7 @@ if (BUILD_SAMPLE)
       device_trezor
       multisig
       version
+      cncrypto
       randomx
       epee
       hardforks

